addon/components/modal-dialog.js

```diff
@@ -14,111 +22,60 @@ const computedJoin = function(prop) {
 };
 
 export default Ember.Component.extend({
-  // modal-dialog is itself tagless. positioned-container provides the container div
   tagName: '',
-
-  layout: template,
-  modalService: injectService('modal-dialog'),
-  destinationElementId: reads('modalService.destinationElementId'),
-
-  useEmberTether: computed('modalService.useEmberTether', 'alignment', 'renderInPlace', function() {
-    return this.get('modalService.useEmberTether') && (this.get('alignment') !== 'none') && !this.get('renderInPlace');
+  layoutStrategy: computed('structure', 'useEmberTether', function() {
+    let strategy = get(this, 'structure');
+    const useEmberTether = get(this, 'useEmberTether');
+    if (strategy === 'tethered' && !useEmberTether) {
```

Comment:
I went back and forth on this, as it feels odd for the 'default' to rely on the optional dependency, then fallback when the dependency isn't present. For what it's worth, this is what's happening in the current versions.. just doesn't feel as weird because it expresses as conditional logic in a single template.